server/src/main/java/io/druid/segment/realtime/RealtimeManager.java

```diff
@@ -325,6 +299,134 @@ public void run()
       }
     }
 
+    private void runFirehoseV2(FirehoseV2 firehose)
+    {
+      final Period intermediatePersistPeriod = config.getIntermediatePersistPeriod();
+      try {
+        firehose.start();
+      }
+      catch (Exception e) {
+      	log.error(e, "Failed to start firehoseV2");
+      	return;
+      }
+      long nextFlush = new DateTime().plus(intermediatePersistPeriod).getMillis();
+      log.info("Firehose started with nextFlush [%s]", nextFlush);
+      boolean haveRow = true;
+      while (haveRow) {
+        InputRow inputRow = null;
+        int numRows = 0;
+        try {
+          inputRow = firehose.currRow();
+          if (inputRow != null) {
+            try {
+              numRows = plumber.add(inputRow);
+            }
+            catch (IndexSizeExceededException e) {
+              log.info("Index limit exceeded: %s", e.getMessage());
+              nextFlush = doIncrementalPersist(firehose.makeCommitter(), intermediatePersistPeriod);
+              continue;
+            }
+
+            if (numRows < 0) {
+              metrics.incrementThrownAway();
+              log.debug("Throwing away event[%s]", inputRow);
+            } else {
+              metrics.incrementProcessed();
+            }
+          }
+        }
+        catch (Exception e) {
```

Comment:
I don't think this is a good idea. I don't think we want exceptions in firehose.makeCommitter, plumber.add, or plumber.persist to cause us to skip over whatever InputRow happened to trigger it, especially in a design that is trying to keep Druid consistent with what is in the Firehose.

My reasoning is that the Plumbers shouldn't be throwing exceptions except in case of bugs in the indexing code or running out of space on disk. In those situations, as someone running Druid, I'd actually want the indexer to send a really loud alert and shut down. I would probably set up some external supervision that tries to re-start the task when that happens, since that way I can try to get it running again from a checkpoint, which is preferred since it doesn't sacrifice the guarantees we're trying to provide.

If people are really demanding this behavior then I think it should be an option that is off by default.